packages/eslint-plugin/src/rules/no-implied-eval.ts

```diff
@@ -0,0 +1,123 @@
+import ts from 'typescript';
+import {
+  TSESTree,
+  AST_NODE_TYPES,
+} from '@typescript-eslint/experimental-utils';
+import * as util from '../util';
+
+const EVAL_LIKE_METHODS = new Set([
+  'setImmediate',
+  'setInterval',
+  'setTimeout',
+  'execScript',
```

Comment:
we should support here also `new Function`
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function

Reply:
I completely forgot that

```ts
Function('console.log(1)')
new Function('console.log(1)')
// probably don't need to cover this case because it's pretty out there
Function.prototype.constructor('console.log(1)')
```

Reply:
@armano2 @bradzacher Thanks for the feedback.

I thought about this case, however, I decided to ignore it in favor of rule https://eslint.org/docs/rules/no-new-func, which completely disallows using `Function`. 
`Function` can accept an array of string (`string[]`) and cannot accept a function and TS handles it. I can add a condition to check if `Function` accepts any arguments then throw an error. Should I do it?

What do you think?

Reply:
That's a good thought, but I think it's probably a good idea to add it here.
It'd be pretty confusing as a user if you only used this rule, then you found it missed `Function`.

I don't have a problem with the overlap / double reporting.